File: intellij-plugin/src/main/java/com/github/tth05/tethintellijplugin/TethLanguageFileType.java
package com.github.tth05.tethintellijplugin;

import com.intellij.icons.AllIcons;
import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.NlsContexts;
import com.intellij.openapi.util.NlsSafe;
import org.jetbrains.annotations.NonNls;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public class TethLanguageFileType extends LanguageFileType {

    public static final TethLanguageFileType INSTANCE = new TethLanguageFileType();

    private TethLanguageFileType() {
        super(TethLanguage.INSTANCE);
    }

    @Override
    public @NonNls @NotNull String getName() {
        return "Teth file";
    }

    @Override
    public @NlsContexts.Label @NotNull String getDescription() {
        return "Teth language file";
    }

    @Override
    public @NlsSafe @NotNull String getDefaultExtension() {
        return "teth";
    }

    @Override
    public Icon getIcon() {
        return AllIcons.Debugger.AddToWatch;
    }
}

File: intellij-plugin/src/main/kotlin/com/github/tth05/tethintellijplugin/TethLanguageFileType.kt
package com.github.tth05.tethintellijplugin

import com.intellij.openapi.fileTypes.LanguageFileType
import com.intellij.openapi.util.NlsContexts
import com.intellij.openapi.util.NlsSafe
import org.jetbrains.annotations.NonNls
import javax.swing.Icon

class TethLanguageFileType : LanguageFileType(TethLanguage.INSTANCE) {

    override fun getName(): String = "Teth File"

    override fun getDescription(): String = "Teth language file"

    override fun getDefaultExtension(): String = "teth"

    override fun getIcon(): Icon = TethIcons.FILE

    companion object {
        val INSTANCE = TethLanguageFileType()
    }
}
